library/common/main_interface.h

```diff
@@ -4,11 +4,21 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include "engine.h"
```

Comment:
Prefer full include paths, not relative. In this case though I think you can just forward declare Engine and avoid this include:

```
namespace Envoy {
class Engine;
}
```